docs/architecture/architecture.rst

```diff
@@ -0,0 +1,92 @@
+..
+    This file is part of Zenodo.
+    Copyright (C) 2018 CERN.
+
+    Zenodo is free software; you can redistribute it
+    and/or modify it under the terms of the GNU General Public License as
+    published by the Free Software Foundation; either version 2 of the
+    License, or (at your option) any later version.
+
+    Zenodo is distributed in the hope that it will be
+    useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Zenodo; if not, write to the
+    Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+    MA 02111-1307, USA.
+
+    In applying this license, CERN does not
+    waive the privileges and immunities granted to it by virtue of its status
+    as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+=============
+ Architecture
+=============
+
+Statistics
+----------
+
+Events
+~~~~~~
+
+The usage statistics of the Zenodo records are generated from two different types of events:
+
+* ``record-view``: this event is related to the view of a record and it's bounded to the ``record_viewed`` signal, which is
+  emitted when one the following endpoints is accessed:
+
+    * ``/record/<recid>``
+    * ``/record/<recid>/export/<format>``
+
+* ``file-download``: this event is related to the download a file of a record and it's is bounded to
+  the  ``file_downloaded`` signal, which is emitted when one the following endpoints is accessed:
+
+    * ``/record/<recid>/files/<filename>``
+    * ``/api/record/<recid>/files/<filename>``
+
+Every time a user views a record or downloads a file, the corresponding signal is emitted and the corresponding
```

Comment:
It's worth mentioning that in case an event fails to be emitted (e.g. an exception is raised in an event builder or sending the message to RabbitMQ fails), the request is not affected. A warning is logged though so that we can examine later e.g. in Sentry